core/src/main/java/hudson/model/FingerprintCleanupThread.java

```diff
@@ -107,7 +107,7 @@ private void deleteIfEmpty(File dir) {
     private boolean check(File fingerprintFile, TaskListener listener) {
         try {
             Fingerprint fp = loadFingerprint(fingerprintFile);
-            if (fp == null || !fp.isAlive()) {
+            if (fp == null || (!fp.isAlive() && !fp.isFingerprintDeletionBlocked()) ) {
```

Comment:
Might be great to add some logging later, but the current chance follows the convention

Reply:
Thanks @oleg-nenashev , I added logging, let me know if it is fine.